### PR TITLE
E2E: Code reduction and todo modification

### LINKED ISSUE #642: feature: e2e code optimisation
(1) Increase BeforeEach reuse rate to reduce the amount of code 
(2) Reduce duplicate api to reduce code size 
(3) Use case timing optimisation

### PR DESCRIPTION
**Pull request by ty-dc**
Signed-off-by: ty-dc [tao.yang@daocloud.io](mailto:tao.yang@daocloud.io)

What type of PR is this?
/kind pr/release/none-required

What this PR does / why we need it:
- (1) Increase BeforeEach reuse rate to reduce the amount of code
- (2) Reduce duplicate api to reduce code size
- (3) Use case timing optimisation

Which issue(s) this PR fixes:
Fixes #642

### CODE REVIEW

**Review by yulng — APPROVED** [review]

**Review by bzsuni — DISMISSED** [review]

File: test/doc/reliability.md [hunk 1]
```diff
@@ -7,5 +7,5 @@
 | R00003  | Successfully run a pod during the API-server is restarting            | p2       |       | done   |       |
```
Comment (windsonsea) [inline test/doc/reliability.md:7]:
```suggestion
| R00003  | Successfully run a pod when the API-server is restarting            | p2       |       | done   |       |
```
Same issue below.

File: test/doc/reliability.md [hunk 2]
```diff
@@ -7,5 +7,5 @@
...
 | R00004  | Successfully run a pod during the Spiderpool agent is restarting      | p2       |       | done   |       |
 | R00005  | Successfully run a pod during the coreDns is restarting               | p3       |       | done   |       |
-| R00006  | Successfully recovery a pod whose original node is power-off          | p2       |       | NA     |       |
+| R00006  | Successfully recovery a pod whose original node is power-off          | p2       |       | done   |       |
```
Comment (windsonsea) [inline test/doc/reliability.md:10]:
```suggestion
| R00006  | Successfully recover a pod whose original node is power-off          | p2       |       | done   |       |
```

File: test/doc/reservedip.md [hunk 3]
```diff
@@ -2,6 +2,6 @@
 
 | Case ID | Title                                                                            | Priority | Smoke | Status | Other |
 |---------|----------------------------------------------------------------------------------|----------|-------|--------|-------|
-| S00001  | An IP that is set in ReservedIP CRD, should not be assigned to a pod              | p2       |       | NA     |       |
+| S00001  | An IP that is set in ReservedIP CRD, should not be assigned to a pod              | p2       |       | done   |       |
```
Comment (windsonsea) [inline test/doc/reservedip.md:5]:
```suggestion
| S00001  | An IP that is set in ReservedIP CRD should not be assigned to a pod              | p2       |       | done   |       |
```

File: test/doc/annotation.md [hunk 4]
```diff
@@ -3,12 +3,12 @@
...
+| A00002  | Added fields such as `"dist":"1.0.0.0/16"`, `"gw":"1.0.0.1"`, and `nics` and the pod was running successfully | p2       |       | done   |       |
 | A00003  | Failed to run a pod with invalid annotations                 | p3       |       | done   |       |
 | A00004  | Take a test with the Priority: pod annotation > namespace annotation > specified in a CNI profile > settings in ConfigMap | p1       |       | NA     |       |
 | A00005  | The "IPPools" annotation has the higher Priority over the "IPPool" annotation | p1       |       | done   |       |
 | A00006  | The namespace annotation has precedence over global default IPPool | p1       | true  | done   |       |
-| A00007  | Spiderpool will successively try to allocate IP in the order of the elements in the IPPool array until the first allocation succeeds or all fail | p1       | true  | NA     |       |
-| A00008  | Successfully run an annotated multi-container pod            | p2       |       | NA     |       |
+| A00007  | Spiderpool will successively try to allocate IP in the order of the elements in the IPPool array until the first allocation succeeds or all fail | p1       | true  | done   |       |
```
Comment (windsonsea) [inline test/doc/annotation.md:11]:
```suggestion
| A00007  | Spiderpool will successively try to allocate IPs in the order of the elements in the IPPool array until the first allocation succeeds or all fail | p1       | true  | done   |       |
```

File: test/doc/affinity.md [hunk 5]
```diff
@@ -2,12 +2,12 @@
...
-| L00003  | Successfully run deployment/pod who is bound to an IPPool set with matched namespaceSelector        | p2       | true  | NA     |       |
-| L00004  | Failed to run deployment/pod who is bound to an IPPool set with no-matched namespaceSelector        | p3       |       | NA     |       |
-| L00005  | Successfully run deployment/pod who is bound to an IPPool set with matched podSelector              | p2       | true  | NA     |       |
-| L00006  | Failed to run deployment/pod who is bound to an IPPool set with no-matched podSelector              | p3       |       | NA     |       |
-| L00007  | Successfully run daemonSet/pod who is cross-zone daemonSet with matched nodeSelector                | p2       |       | NA     |       |
-| L00008  | Successfully restarted statefulSet/pod with matching podSelector, ip remains the same               | p2       |       | NA     |       |
+| L00001  | Successfully run deployment/pod who is bound to an IPPool set with matched `NodeAffinity`           | p2       | true  | done   |       |
```
Comment (windsonsea) [inline test/doc/affinity.md:5]:
```suggestion
| L00001  | Successfully run deployment/pod that is bound to an IPPool set with matched `NodeAffinity`           | p2       | true  | done   |       |
```
If possible, `who` is used for an object with life and `that` for everything. Same below.

File: .github/workflows/auto-pr-ci.yaml [hunk 6]
```diff
@@ -374,9 +374,9 @@ jobs:
           echo ${{ github.event.inputs.labels }}
           LABELS=
           if [ -n "${{ needs.get_ref.outputs.e2e_labels }}" ]; then
-              LABELS=" ${{ needs.get_ref.outputs.e2e_labels }}  && !performance "
+              LABELS=" ${{ needs.get_ref.outputs.e2e_labels }}  && !performance && !reliability "
           else
-              LABELS=" !performance "
+              LABELS=" !performance && !reliability "
```
Comment (weizhoublue) [inline .github/workflows/auto-pr-ci.yaml:379]: 如果 提交了一个 测试用例，其关于 reliability 的 ，那么 岂不是 测不到其  单栈  的代码是否正确了

**Review by weizhoublue — APPROVED** [review]

**Pull request merged**